src/trap.cpp

```diff
@@ -201,7 +201,11 @@ bool trap::detected_by_ground_sonar() const
     return loadid == tr_beartrap_buried || loadid == tr_landmine_buried || loadid == tr_sinkhole;
 }
 
-bool trap::detect_trap( const tripoint &pos, const Character &p ) const
+/** The base vision score that is checked against trap visibility to see if it's detected.
+ * This, _base_ function does not factor in the random chance, or the relative distance between
+ * the player and the trap (which *are* factors in trap detection).
+ */
+int trap_base_detection_score( const Character &p )
```

Comment:
This feels very much like it should be a method on `Character`, but 1) it's only used in the function below (and tests) and 2) I think it's a good idea to keep the explanatory comments close to `trap::detect_trap`